pkg/apis/hive/v1/clusterdeployment_types.go

```diff
@@ -128,6 +128,10 @@ type ClusterDeploymentSpec struct {
 	// time since the cluster last came out of hibernation.
 	// +optional
 	HibernateAfter *metav1.Duration `json:"hibernateAfter,omitempty"`
+
+	// InstallAttemptsLimit is the maximum number of times Hive will attempt to install the cluster.
+	// +optional
+	InstallAttemptsLimit *int32 `json:"installAttemptsLimit,omitempty"`
```

Comment:
is `0` is valid value? I don't think *int32 is required here, using int32 and using >0 for when this is used
- keeps the API backwards compatible in since 0 will be the default value of decoded int32
- keeps the API much simple as users don't have to capture nil

Reply:
`0` **is** a valid value. It means that the user does not want to attempt any installs. A nil value means that the user does not want to impose any limit on the number of install attempts.

Reply:
The code is comparing this value to cd.Status.InstallRestarts , and restart == 0 doesn't mean there was no attempt (if i go by the name). So the handling should be updated to make sure the controller never starts provisioning..

Reply:
restart == 0 does mean there was no attempt before. We increment installRestarts after previous failed attempt and before starting a new provision.

Reply:
> restart == 0 does mean there was no attempt before. We increment installRestarts after previous failed attempt and before starting a new provision.

that's not what `InstallRestarts` name means, but as long as it does what you say it does.